app/models/user/reputation_tokens/mentored_token.rb

```diff
@@ -6,6 +6,10 @@ class User::ReputationTokens::MentoredToken < User::ReputationToken
   reason :mentored
   value 5
 
+  before_validation on: :create do
+    self.earned_on = self.discussion.finished_at unless earned_on
```

Comment:
We might need to think about this, as reputation is actually probably awarded when the testimonial is left. But this is fine for now.

Reply:
Aha. Good to come back at.